bot/app/database/db_objects.py

```diff
@@ -0,0 +1,214 @@
+""" Do Do :
+
+playsAt:
+- insert functionality into Event and Artist Classes
+
+all:
+- fix documentation strings in functions to include: 
+    - what database item
+    - which operations incl. parameters
+- add logging, catch errors to log
+"""
+
+import sqlite3
+import logging
+import os
+
+logging.basicConfig(os.path.abspath('../log.txt', level=logging.DEBUG))
+
+DB_NAME = 'data.sqlite'
+
+class Event:
+
+    """
+    An instance of the Event relation 
+    """
+
+    def __init__(self, e_name, date, time, admi, desc, loca, e_pic_id):
+        
+        
+        self.e_name = e_name
+        self.date = date
+        self.time = time
+        self.admi = admi
+        self.desc = desc
+        self.loca = loca
+        self.e_pic_id = e_pic_id
+
+        self.insert_event()
+    
+    def insert_event(self):
+        
+        conn = sqlite3.connect(DB_NAME)
+        curs = conn.cursor()
+
+        try:
+            curs.execute("""
+                insert into Events values (?,?,?,?,?,?)""",
+                (self.e_name, self.date, self.time, self.admission, self.desc, \
+                self.e_pic_id))
+            self.conn.commit()
+        except sqlite3.IntegrityError as e:
+            logging.error(e)
+        
+        curs.close()
+        conn.close()
+        
+class Artist:
+
+    def __init__(self, a_name, webs, soundc, bandc, bio, a_pic_id):
+
+        self.a_name = a_name
+        self.webs = webs
+        self.soundc = soundc
+        self.bandc = bandc
+        self.bio = bio
+        self.a_pic_id = a_pic_id
+
+        self.insert_artist()
+        
+    def insert_artist(self):
+        
+        conn = sqlite3.connect(DB_NAME)
+        curs = conn.cursor()
+
+        try:
+            curs.execute("""
+                insert into Artists values (?,?,?,?,?,?)""",
+                (self.a_name, self.webs, self.soundc, self.bandc, self.bio, \
+                self.a_pic_id))
+            self.conn.commit() 
+        except sqlite3.IntegrityError as e:
+            logging.error(e)
+        
+        curs.close()
+        conn.close()
+
+class PlaysAt:
+
+    """
+    Instance represents Artist-Event connection (aka playsAt Table)
+    """
+
+    def __init__(self):
+        
+        self.a_name = input("Enter the Artist's name: ")
+        self.e_name = input("Enter the Event's name: ")
+        self.date = input("Enter the Event's date (format DD.MM.YYYY): ")
+
+        self.check_for_artist()
+        self.check_for_event()
+        self.insert_plays_at()
```

Comment:
I would again decouple this from the constructor.

Reply:
fixed